File: two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> index; 
        vector<int> ans;
        int i;
        for(i = 0; i < nums.size(); i++) {
            if(index.find(nums[i]) != index.end()) {
                break;
            }
            index[target - nums[i]] = i;
        }
        ans.push_back(index[nums[i]]);
        ans.push_back(i);
        return ans;
    }
};

File: two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        index = {}
        for idx, num in enumerate(nums):
            if num in index:
                break
            index[target - num] = idx
        return [index[num], idx]